### PR TITLE
[SMTChecker] Refactor expressions to use SymbolicVariable

### PR DESCRIPTION
**Pull request by leonardoalt**
Depends on #5235

### CODE REVIEW

**Review by chriseth — DISMISSED** [review]
Reviewed the last commit.

**Comment by leonardoalt** [comment]
@chriseth Just rebased after merging the other PR, can you approve again?

**Comment by leonardoalt** [comment]
But don't merge yet, need to check something.

File: libsolidity/formal/SMTChecker.cpp [hunk 1]
```diff
@@ -996,34 +990,15 @@ bool SMTChecker::hasVariable(VariableDeclaration const& _var) const
...
-	case Type::Category::Function:
-		// This should be replaced by a `non-deterministic` type in the future.
-		m_expressions.emplace(&_e, m_interface->newInteger(exprSymbol));
-		break;
-	default:
-		m_expressions.emplace(&_e, m_interface->newInteger(exprSymbol));
+	if (hasExpr(_e))
+		m_expressions.at(&_e)->increaseIndex();
+	auto result = newSymbolicVariable(*_e.annotation().type, "expr_" + to_string(_e.id()), *m_interface);
+	m_expressions.emplace(&_e, result.second);
```
Comment (leonardoalt) [inline libsolidity/formal/SMTChecker.cpp]: This should be inside `else` otherwise it has no effect.

**Comment by leonardoalt** [comment]
Ok, ready for review/merge again.

**Review by chriseth — APPROVED** [review]

**Pull request merged**